packages/desktop-client/src/components/sidebar/BudgetName.tsx

```diff
@@ -0,0 +1,169 @@
+        navigate('/settings');
+        break;
+      case 'close':
+        dispatch(closeBudget());
+        break;
+      default:
+    }
+  }
+
+  const items = [
+    { name: 'settings', text: t('Settings') },
+    { name: 'close', text: t('Close file') },
+  ];
+
+  return (
+    <View
+      style={{
+        padding: '15px 8px',
+        flexDirection: 'row',
+        alignItems: 'center',
+        userSelect: 'none',
+        transition: 'padding .4s',
+        ...(hasWindowButtons && {
+          paddingTop: 0,
+          justifyContent: 'flex-start',
+        }),
+        '& .hover-visible': {
+          opacity: 0,
+          transition: 'opacity .25s',
+        },
+        '&:hover .hover-visible': {
+          opacity: 1,
+        },
+      }}
+    >
+      <Button
+        ref={triggerRef}
+        variant="bare"
+        style={{
+          margin: 5,
+          flex: '0 auto',
+        }}
+        onPress={() => setMenuOpen(true)}
+      >
+        <SvgLogo
+          width={20}
+          height={20}
+          style={{
+            color: theme.sidebarItemTextSelected,
+          }}
+        />
+      </Button>
+
+      <Popover
+        triggerRef={triggerRef}
+        placement="bottom start"
+        isOpen={menuOpen}
+        onOpenChange={() => setMenuOpen(false)}
+      >
+        <Menu onMenuSelect={onMenuSelect} items={items} />
+      </Popover>
+
+      <EditableBudgetName />
+
+      <View style={{ flex: 1, flexDirection: 'row' }} />
+
+      {children}
+    </View>
+  );
+}
+
+function EditableBudgetName() {
+  const { t } = useTranslation();
+  const [budgetName, setBudgetNamePref] = useMetadataPref('budgetName');
+  const [editing, setEditing] = useState(false);
+
+  if (editing) {
+    return (
+      <InitialFocus>
+        <Input
+          style={{
+            maxWidth: 'calc(100% - 73px)',
+            fontSize: 16,
+            fontWeight: 500,
+          }}
+          defaultValue={budgetName}
+          onEnter={e => {
+            const inputEl = e.target as HTMLInputElement;
+            const newBudgetName = inputEl.value;
+            if (newBudgetName.trim() !== '') {
+              setBudgetNamePref(newBudgetName);
+              setEditing(false);
+            }
+          }}
+          onBlur={() => setEditing(false)}
+        />
+      </InitialFocus>
+    );
+  }
+
+  return (
+    <>
+      <Text
+        style={{
+          whiteSpace: 'nowrap',
+          overflow: 'hidden',
+          fontSize: 16,
+          fontWeight: 500,
+        }}
+      >
+        {budgetName || t('A budget has no name')}
+      </Text>
+
+      <Button
+        variant="bare"
+        aria-label={t('Edit budget name')}
+        className="hover-visible"
+        onPress={() => setEditing(true)}
+        style={{ margin: 5 }}
+      >
+        <SvgPencil1
+          style={{
+            width: 11,
+            height: 11,
+            color: theme.pageTextSubdued,
+          }}
+        />
+      </Button>
```

Comment:
It would be good to have better accessibility, especially for touch only devices.  However, ```@media``` is not implemented on ```<Button>``` at this time.  This will need to be implemented in the future.